### PR TITLE
add vitest support

### PR DESCRIPTION
**Pull request by silkentrance**
add vitest support
v0.3.0

### CODE REVIEW

**Comment by silkentrance** [comment]
CallbackSuite generates an error

```
@testdeck/vitest:  FAIL  test.ts > tests > CallbacksSuite
@testdeck/vitest: TypeError: Callback must be a function. Received <ref *1> {
```

where a `vitest` suite object is passed as a callback.

**Comment by silkentrance** [comment]
While this builds and works, the available tests and also the necessary deprecation of node v15.x (EOL), bothers me a lot.

**Comment by silkentrance** [comment]
rebased to current main and also bumped version to v0.3.0 since we are adding a new feature

**Review by pana-cc — APPROVED** [review]
There are few more things to do - check the README.md.

About the timeout - maybe we can calculate the timeouts and somehow store them in the LifecycleSettings:
https://github.com/testdeck/testdeck/blob/78136864cc34bd30c67c108a3d9551c0783cf669/packages/core/index.ts#L70-L85

I guess we could add `suiteTimeout` and `suiteSlow` fields, and get them there from `obj.constructor[ClassTestUI.timeoutSymbol]`, then let the test interfaces decide whether to use the classTimeout if a test timeout is not set. I think this would be a bit cleaner than passing it along the stack. The stack would work, but hopefully one day we could execute parallel tests in the same context.

**Comment by pana-cc** [comment]
BTW, I would be happy to help on this PR. I don't mind dropping node 15. It is not an LTS anyway.

File: packages/vitest/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+## @testdeck/jest
+
+[![Get it on NPM](https://img.shields.io/npm/v/@testdeck/jest.svg)](https://www.npmjs.com/package/@testdeck/jest)
```
[the remark this answers is not in the document]
Comment (silkentrance) [reply packages/vitest/README.md]: done

**Comment by silkentrance** [comment]
@pana-cc I do not understand and how this is relevant to adding support for vitest?

```
About the timeout - maybe we can calculate the timeouts and somehow store them in the LifecycleSettings:
```

Can you please elaborate on this?

**Comment by panayot-cankov** [comment]
`About the timeout - maybe we can calculate the timeouts and somehow store them in the LifecycleSettings:` Nevermind. I think I read some sort of "TODO" that for @timeout on a class to work we have to put it in a stack for the @test to be able to reference it. If my suggestion doesn't ring a bell - ignore it.

**Comment by silkentrance** [comment]
> BTW, I would be happy to help on this PR. I don't mind dropping node 15. It is not an LTS anyway.

@pana-cc Feel free. I will no longer force push commits on this branch.

I for my part am mostly done. There are a few tests that I would like to add, however, these could be added to the other packages as well. I think it is best to do this in a different PR.

**Comment by silkentrance** [comment]
> `About the timeout - maybe we can calculate the timeouts and somehow store them in the LifecycleSettings:` Nevermind. I think I read some sort of "TODO" that for @timeout on a class to work we have to put it in a stack for the @test to be able to reference it. If my suggestion doesn't ring a bell - ignore it.

Found it: testdeck/jest/index.ts:

```
  5             // TODO: Push settings.timeout on the stack and pass it to each child test...
```

However, for this we need to investigate further.

**Comment by silkentrance** [comment]
requires #317

**Comment by silkentrance** [comment]
I squashed all commits and reverted versions to 0.2.1 so that we can use the lerna-publish workflow

**Comment by silkentrance** [comment]
rebased to current main

**Pull request merged**